packages/store/src/IStoreHook.sol

```diff
@@ -8,16 +8,18 @@ import { PackedCounter } from "./PackedCounter.sol";
 // ERC-165 Interface ID (see https://eips.ethereum.org/EIPS/eip-165)
 bytes4 constant STORE_HOOK_INTERFACE_ID = IStoreHook.onBeforeSetRecord.selector ^
   IStoreHook.onAfterSetRecord.selector ^
-  IStoreHook.onBeforeSetField.selector ^
-  IStoreHook.onAfterSetField.selector ^
+  IStoreHook.onBeforeSpliceStaticData.selector ^
+  IStoreHook.onAfterSpliceStaticData.selector ^
+  IStoreHook.onBeforeSpliceDynamicData.selector ^
+  IStoreHook.onAfterSpliceDynamicData.selector ^
   IStoreHook.onBeforeDeleteRecord.selector ^
   IStoreHook.onAfterDeleteRecord.selector ^
   ERC165_INTERFACE_ID;
 
 interface IStoreHook is IERC165 {
   function onBeforeSetRecord(
     bytes32 tableId,
-    bytes32[] memory keyTuple,
+    bytes32[] calldata keyTuple,
```

Comment:
will we have similar issues as https://github.com/latticexyz/mud/pull/1528#discussion_r1328132000 by changing this?

Reply:
Changed all to `memory`. Had `calldata` initially bc it would be cheaper if called with calldata, but we're in fact calling all these functions with memory from the `World`, so needs to be memory here.